tfx/orchestration/pipeline.py

```diff
@@ -189,3 +190,76 @@ def components(self, components: List[base_node.BaseNode]):
     # has all its dependencies visited.
     if len(self._components) < len(deduped_components):
       raise RuntimeError('There is a cycle in the pipeline')
+
+  def get_fuseable_components(self):
+    """
+       Returns a list of fuseable Beam component subraphs in sorted topological
+       order. For example:
+
+       [[component_a, component_b],[component_c, component_d, component_e]]
+
+       Experimental, backward compatability not guaranteed.
+    """
+
+    # Finds all Beam based components
+    beam_components = set()
+
+    for component in self.components:
+      if issubclass(component.EXECUTOR_SPEC.executor_class,
+                    base_executor.FuseableBeamExecutor):
+        beam_components.add(component)
+
+    # Finds Beam component "sources" (components with no parents that are Beam
+    # based components).
+    beam_component_sources = []
+
+    for component in sorted(beam_components, key=lambda c: c.id):
+      is_source = True
+
+      for parent in component.upstream_nodes:
+        if parent in beam_components:
+          is_source = False
+          break
+
+      if is_source:
+        beam_component_sources.append(component)
+
+    # Finds subgraphs of fuseable beam components, sorted in topological order
+    fuseable_components = []
```

Comment:
Okay, this is somewhat confusing. There's already a variable named `beam_components`, which is really "components that have the fuseable executor". And there's `fuseable_components`, which is really "list of subgraphs, in which each subgraph can be fused into one component". Could we pick better names to disambiguate these two variables?

Reply:
I agree. I replaced `beam_components` with the function `_is_beam_component()`.  Also, `fuseable_components` was changed to `fuseable_subgraphs`. Some other naming was changed as well to follow this convention.